src/System.Linq.Expressions/src/System/Runtime/CompilerServices/CallSite.cs

```diff
@@ -667,7 +667,7 @@ private T CreateCustomNoMatchDelegate(MethodInfo invoke)
                 Expression.Block(
                     Expression.Call(
                         typeof(CallSiteOps).GetMethod(nameof(CallSiteOps.SetNotMatched)),
-                        @params.First()
+                        @params[0]
```

Comment:
If I'm right about that being the only path, maybe throw in an assertion, all the same.